l5kit/l5kit/data/filter.py

```diff
@@ -83,22 +83,35 @@ def filter_agents_by_track_id(agents: np.ndarray, track_id: int) -> np.ndarray:
     return agents[np.nonzero(agents["track_id"] == track_id)[0]]
 
 
-def get_agent_by_track_id(all_agents: np.ndarray, frame: np.ndarray, track_id: int) -> Optional[np.ndarray]:
+def get_agent_by_track_id(agents_frame: np.ndarray, track_id: int) -> Optional[np.ndarray]:
     """Return the agent object (np.ndarray) of a given track_id in a frame.
     Return None if the agent is not among those in the frame.
 
     Arguments:
-        all_agents (np.ndarray): agents array
-        frame (np.ndarray): given frame
+        agents_frame (np.ndarray): frame agents array
         track_id (int): agent track id to select
 
     Returns:
         Optional[np.ndarray] -- Selected agent, or None if this agent is not present in given frame.
     """
 
-    agents = filter_agents_by_frame(all_agents, frame)
     try:
-        agent = filter_agents_by_track_id(agents, track_id)[0]
+        agent = filter_agents_by_track_id(agents_frame, track_id)[0]
         return agent
     except IndexError:  # no agent for track_id in this frame
         return None
+
+
+def get_frames_agents(frames: np.ndarray, agents: np.ndarray) -> List[np.ndarray]:
```

Comment:
@lucabergamini So what this function does is: given all the agents and frames, return only the agents from given frames?

Reply:
@lucabergamini I would make the docs more clear and have a better function name, otherwise LGTM

Reply:
> @lucabergamini I would make the docs more clear and have a better function name, otherwise LGTM
> `filter_agents_by_frames`? be consistent with other funcion name `filter_agents_by_frame`

done. Note that now the `filter_agents_by_frame` is not used in the code...we may want to remove it in the future @yye-lyft